### ISSUE TITLE
Fix `declaration-block-no-redundant-longhand-properties` autofix for `grid-template`

### ISSUE THREAD

**Issue by shoonia** [stylelint/stylelint#6701]
### What steps are needed to reproduce the bug?

Broken autofix for `declaration-block-no-redundant-longhand-properties` with `grid-template`

I have the next CSS
DEMO: https://codepen.io/shoonia/pen/qBMVPXr
```css
.page {
  display: grid;
  height: 100vh;
  grid-template-rows: var(--header-h) 1fr var(--footer-h);
  grid-template-columns: var(--toolbar-w) 1fr;
  grid-template-areas: "header header" "toolbar main" "footer footer";
}
```

After using autofix with `--fix` , it broken my layout
DEMO: https://codepen.io/shoonia/pen/GRXOMMG
```diff
.page {
  display: grid;
  height: 100vh;
-  grid-template-rows: var(--header-h) 1fr var(--footer-h);
-  grid-template-columns: var(--toolbar-w) 1fr;
-  grid-template-areas: "header header" "toolbar main" "footer footer";
+ grid-template: var(--toolbar-w) 1fr var(--header-h) 1fr var(--footer-h) "header header" "toolbar main" "footer footer";
}
```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-standard",
    "stylelint-config-css-modules"
  ],
  "rules": {
    "selector-class-pattern": null,
    "selector-pseudo-element-no-unknown": [
      true,
      {
        "ignorePseudoElements": [
          "range-track",
          "range-thumb"
        ]
      }
    ]
  }
}

```

### How did you run Stylelint?

`npx stylelint src/**/*.css --fix`

### Which version of Stylelint are you using?

`15.2.0`

### What did you expect to happen?

it shouldn't be minimized.

### What actually happened?

```diff
-  grid-template-rows: var(--header-h) 1fr var(--footer-h);
-  grid-template-columns: var(--toolbar-w) 1fr;
-  grid-template-areas: "header header" "toolbar main" "footer footer";
+ grid-template: var(--toolbar-w) 1fr var(--header-h) 1fr var(--footer-h) "header header" "toolbar main" "footer footer";
```

### Does the bug relate to non-standard syntax?

NO

### Proposal to fix the bug

Looks like it shouldn't be minimize. I don't know a short syntax for this construction.

**Comment by jeddy3** [comment]
@shoonia Thanks for the detailed report and for using the template.

I can reproduce [with the demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgcwCcBLKAWhxgFsAHAGyQtIIgHcBneFANyQIApSpABYwksAsICUKAIwAzAt14DSciBAoShkgNxYUKQiXJU6DGKUi0ArpUwcl-QTnW0ARr1Itp8gnswGRmQUNPSMvKIO6CAiYjCKseLRKNEuEO68KJRIRNggKSBqGvEoRZrR-gC+IAA0hUS0MABySJRwiDAAHq10MAB0YGxsteAQmHJEeAggGAEFXRSYUGzRnADa+gYFbDgAno20uTiWYxN4pDtIS7xQ0ZsAuljVdZDjkwBiEATZONMAVmwxiNYNRhohZgZojt9jBDpgcKsCmEYDtojVNlC9gcjic3udLtcCLcEEjzKiQE8QJUgA):

```css
a {
  grid-template-rows: var(--header-h) 1fr var(--footer-h);
  grid-template-columns: var(--toolbar-w) 1fr;
  grid-template-areas: "header header" "toolbar main" "footer footer";
}
```

The autofixed shorthand is incorrect. It should be:

```css
a {
  grid-template: 
      "header header" var(--header-h) 
      "toolbar main" 1fr
      "footer footer" var(--footer-h)
      / var(--toolbar-w) 1fr;
}
```

(As an aside, I think it's an excellent shorthand as you get to see the sizes of the rows and columns in situ.)

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

In the meantime, you can [ignore this shorthand](https://stylelint.io/user-guide/rules/declaration-block-no-redundant-longhand-properties/#ignoreshorthands-regex-regex-string) using:

```json
{
  "rules": {
    "declaration-block-no-redundant-longhand-properties": [true, { "ignoreShorthands": ["grid-template"] }]
  }
}
```

**Comment by mattxwang** [comment]
Ah, I think I overlooked this when implementing the original autofix - that's on me 😓 . The naive approach I used (shuffle the order of the values based on the "correct" property order) doesn't work in this case.

The suspect line is this:

https://github.com/stylelint/stylelint/blob/964b1ba71ac3526a1b3fb305e72c3af3fa3c9e24/lib/rules/declaration-block-no-redundant-longhand-properties/index.js#L116-L120

I can take a more in-depth look and submit a patch to fix in a week or so? But, if someone else wants to submit a patch first, go for it!
